### DISCUSSION TITLE
WiX V3 wixproj converted to WiX V4 wixproj HEAT questions

### DISCUSSION THREAD

**Discussion by MarkStega** [wixtoolset/issues#7296] [Questions] [answered]
1) I was a little surprised to find that after converting my V3 project to V4 that I was still seeing V3 HEAT output
I found that the V4 project file still had
```
<PropertyGroup>
		<PreBuildEvent>
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)CommunicationService.2012\bin" -dr APPLICATIONROOTDIRECTORY1  -cg ServiceComponents1 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.CommunicationService.TargetDir" -out "$(SolutionDir)WiX\HeatService1.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)FileCleanupService\bin" -dr APPLICATIONROOTDIRECTORY2  -cg ServiceComponents2 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.FileCleanupService.TargetDir" -out "$(SolutionDir)WiX\HeatService2.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)LCIFileInterfaceService\bin" -dr APPLICATIONROOTDIRECTORY3  -cg ServiceComponents3 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.LCIFileInterfaceService.TargetDir" -out "$(SolutionDir)WiX\HeatService3.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)PATFileInterfaceService\bin" -dr APPLICATIONROOTDIRECTORY4  -cg ServiceComponents4 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.PATFileInterfaceService.TargetDir" -out "$(SolutionDir)WiX\HeatService4.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)PersistentStorageService\bin" -dr APPLICATIONROOTDIRECTORY5  -cg ServiceComponents5 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.PersistentStorageService.TargetDir" -out "$(SolutionDir)WiX\HeatService5.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)ReportService\bin" -dr APPLICATIONROOTDIRECTORY6  -cg ServiceComponents6 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.ReportService.TargetDir" -out "$(SolutionDir)WiX\HeatService6.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)RIAOFileInterfaceService\bin" -dr APPLICATIONROOTDIRECTORY7  -cg ServiceComponents7 -gg -g1 -sfrag -srd -scom -sreg -template fragment -var "var.RIAOFileInterfaceService.TargetDir" -out "$(SolutionDir)WiX\HeatService7.wxs"  -t $(ProjectDir)Filter.xslt
			call "$(wix)\bin\heat.exe" dir "$(SolutionDir)SystemMonitorService\bin" -dr APPLICATIONROOTDIRECTORY8  -cg ServiceComponents8 -gg -g1 -pog:Binaries -pog:Content -sfrag -srd -scom -sreg -template fragment -var "var.SystemMonitorService.TargetDir" -out "$(SolutionDir)WiX\HeatService8.wxs"  -t $(ProjectDir)Filter.xslt
		</PreBuildEvent>
	</PropertyGroup>
```
This was easily removed.

I added a new itemgroup (just to get started) with a single HarvestDirectory
```
	<ItemGroup>
		<HarvestDirectory Include="..\CommunicationService.2012\CommunicationService.csproj"
						HarvestDirectoryTransforms=".\Filter.xslt"
						ProjectOutputGroups="Binaries;Content;Satellites" />
	</ItemGroup>
```
Finally questions:
1) I don't see a property to direct the output (which is showing up in obj\\_CommunicationService.wxs)

2) The generated wxs has a reference to CommunicationService.exe and CommunicationService.exe.config. However none of the dll's in the project output are included. Is that expected and just something different than the WiX3 generated file or am I missing another parameter?

3) Is there a 'conversion' document that shows what the old switches in a HEAT invocation correspond to in V4?

**Comment by rseanhall** [comment, marked as answer]
1. The output file is automatically included in the build so it shouldn't matter what the name is.
2. HarvestDirectory is supposed to point to a directory not a project file. You might have enabled project harvesting where project references are harvested which is where that file is coming from.
3. You should migrate your direct invocations of `heat.exe dir` to `HarvestDirectory` elements in v3 to make sure you get the results you are expecting. Then they should just work when the project is migrated to v4. The documentation on v3 `HarvestDirectory` is at https://wixtoolset.org/docs/v3/msbuild/target_reference/harvestdirectory/. If it's not obvious how to map the switches then you'd have to look at the source code because I don't think there's documentation linking the switches to the MSBuild parameters.

**Reply by MarkStega** [reply]
Sean,

Sorry the issue was that I started trying to use HarvestProject before I discovered the documentation line that says it shouldn't be used as it is unreliable. And when I tried to modify the question I missed the csproj.

I'd suggest moving the 'DoNotUse' warning to the top of the document. I took a quick look and can't find the repo for the new documentation. I'd happily make changes/clarifications and do a PR if that is possible.

**Reply by rseanhall** [reply]
The v4 documentation is at https://wixtoolset.org/docs/tools/heat/. The v3 documentation doesn't really matter any more but its source is at https://github.com/wixtoolset/wix3/tree/develop/src/chm/documents/msbuild/target_reference and is built to https://github.com/wixtoolset/web/tree/master/src/Docusaurus/versioned_docs/version-v3/msbuild/target_reference.

**Comment by MarkStega** [comment]
@rseanhall I have been using the V4 documentation, but what I was actually asking was for the location of the documentation repository so that I could update and submit pull requests...

**Reply by barnson** [reply]
Every page has an `Edit this page` link that points to the file in the `Web` repo.

**Reply by MarkStega** [reply]
@barnson @rseanhall 
Indeed there is an edit link which brings me to the source repository. However, when I try to create a fork of wixtoolset/web so that I can edit and then create a PR I receive an error of '500'.

**Reply by barnson** [reply]
GitHub's having issues: https://twitter.com/githubstatus/status/1640340056815673345

**Reply by MarkStega** [reply]
@barnson -- You were correct, it was the widespread github failure; Creating a fork works as expected now.
